### DISCUSSION TITLE
2 ERRORS!

### DISCUSSION THREAD

**Discussion by zyzy5** [xxyzz/WordDumb#257] [Q&A]
calibre, version 7.15.0 (win32, embedded-python: True)
Tonnerre de Brest!: 发生了一个错误，请复制错误信息并到 GitHub 提交错误报告。

Starting job: 正在为《Theorising Narrative (Un)reliability: A Tentative Roadmap》生成X-Ray 
Job: "正在为《Theorising Narrative (Un)reliability: A Tentative Roadmap》生成X-Ray" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 85, in start_work
  File "calibre_plugins.worddumb.parse_job", line 177, in do_job
  File "calibre_plugins.worddumb.deps", line 53, in install_deps
  File "calibre_plugins.worddumb.deps", line 138, in pip_install
  File "calibre_plugins.worddumb.utils", line 50, in run_subprocess
  File "subprocess.py", line 571, in run
subprocess.CalledProcessError: Command '['C:\\Users\\h5482\\Downloads\\pyenv\\pyenv-win-master\\pyenv-win\\versions\\3.11.8-win32\\python311.exe', '-m', 'pip', '--disable-pip-version-check', 'install', '-U', '-t', 'C:\\Users\\h5482\\AppData\\Roaming\\calibre\\plugins\\worddumb-libs-py3.11', '--no-user', '--no-cache-dir', 'https://github.com/explosion/spacy-models/releases/download/en_core_web_md-3.7.1/en_core_web_md-3.7.1-py3-none-any.whl']' returned non-zero exit status 1.
 
Called with args: (ParseJobData(book_id=107, book_path='C:\\Users\\h5482\\Calibre1997\\Nunning, Vera\\Theorising Narrative (Un)reliabilit (107)\\Theorising Narrative (Un)reliab - Nunning, Vera_x_ray.epub', mi=<calibre.ebooks.metadata.book.base.Metadata object at 0x000002045F46A050>, book_fmt='EPUB', book_lang='en', useragent='WordDumb/3.33.0 (https://github.com/xxyzz/WordDumb)', plugin_path=WindowsPath('C:/Users/h5482/AppData/Roaming/calibre/plugins/WordDumb.zip'), spacy_model='en_core_web_md', create_ww=False, create_x=True, asin='', acr='', revision='', kfx_json=None, mobi_html=b'', mobi_codec=''),) {'notifications': <queue.Queue object at 0x0000020456C9F350>, 'abort': <threading.Event at 0x2045c1d7250: unset>, 'log': <calibre.utils.logging.GUILog object at 0x000002045C1D7090>} 
  error: subprocess-exited-with-error
  
  pip subprocess to install build dependencies did not run successfully.
  exit code: 1
  
  [122 lines of output]
  Ignoring numpy: markers 'python_version < "3.9"' don't match your environment
  Collecting setuptools
    Using cached setuptools-74.1.0-py3-none-any.whl.metadata (6.7 kB)
  Collecting cython<3.0,>=0.25
    Using cached Cython-0.29.37-py2.py3-none-any.whl.metadata (3.1 kB)
  Collecting cymem<2.1.0,>=2.0.2
    Using cached cymem-2.0.8.tar.gz (9.8 kB)
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Collecting preshed<3.1.0,>=3.0.2
    Using cached preshed-3.0.9.tar.gz (14 kB)
    Installing build dependencies: started
    Installing build dependencies: finished with status 'error'
    error: subprocess-exited-with-error
  
    pip subprocess to install build dependencies did not run successfully.
    exit code: 1
  
    [88 lines of output]
    Collecting setuptools
      Using cached setuptools-74.1.0-py3-none-any.whl.metadata (6.7 kB)
    Collecting cython>=0.28
      Using cached Cython-3.0.11-cp311-cp311-win32.whl.metadata (3.2 kB)
    Collecting cymem<2.1.0,>=2.0.2
      Using cached cymem-2.0.8.tar.gz (9.8 kB)
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    Collecting murmurhash<1.1.0,>=0.28.0
      Using cached murmurhash-1.0.10.tar.gz (12 kB)
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    Using cached setuptools-74.1.0-py3-none-any.whl (1.3 MB)
    Using cached Cython-3.0.11-cp311-cp311-win32.whl (2.6 MB)
    Building wheels for collected packages: cymem, murmurhash
      Building wheel for cymem (pyproject.toml): started
      Building wheel for cymem (pyproject.toml): finished with status 'error'
      error: subprocess-exited-with-error
  
      Building wheel for cymem (pyproject.toml) did not run successfully.
      exit code: 1
  
      [17 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win32-cpython-311
      creating build\lib.win32-cpython-311\cymem
      copying cymem\about.py -> build\lib.win32-cpython-311\cymem
      copying cymem\__init__.py -> build\lib.win32-cpython-311\cymem
      creating build\lib.win32-cpython-311\cymem\tests
      copying cymem\tests\test_import.py -> build\lib.win32-cpython-311\cymem\tests
      copying cymem\tests\__init__.py -> build\lib.win32-cpython-311\cymem\tests
      copying cymem\cymem.pyx -> build\lib.win32-cpython-311\cymem
      copying cymem\cymem.pxd -> build\lib.win32-cpython-311\cymem
      copying cymem\__init__.pxd -> build\lib.win32-cpython-311\cymem
      running build_ext
      building 'cymem.cymem' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]
  
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for cymem
      Building wheel for murmurhash (pyproject.toml): started
      Building wheel for murmurhash (pyproject.toml): finished with status 'error'
      error: subprocess-exited-with-error
  
      Building wheel for murmurhash (pyproject.toml) did not run successfully.
      exit code: 1
  
      [22 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win32-cpython-311
      creating build\lib.win32-cpython-311\murmurhash
      copying murmurhash\about.py -> build\lib.win32-cpython-311\murmurhash
      copying murmurhash\__init__.py -> build\lib.win32-cpython-311\murmurhash
      creating build\lib.win32-cpython-311\murmurhash\tests
      copying murmurhash\tests\test_hash.py -> build\lib.win32-cpython-311\murmurhash\tests
      copying murmurhash\tests\test_import.py -> build\lib.win32-cpython-311\murmurhash\tests
      copying murmurhash\tests\__init__.py -> build\lib.win32-cpython-311\murmurhash\tests
      copying murmurhash\mrmr.pyx -> build\lib.win32-cpython-311\murmurhash
      copying murmurhash\mrmr.pxd -> build\lib.win32-cpython-311\murmurhash
      copying murmurhash\__init__.pxd -> build\lib.win32-cpython-311\murmurhash
      creating build\lib.win32-cpython-311\murmurhash\include
      creating build\lib.win32-cpython-311\murmurhash\include\murmurhash
      copying murmurhash\include\murmurhash\MurmurHash2.h -> build\lib.win32-cpython-311\murmurhash\include\murmurhash
      copying murmurhash\include\murmurhash\MurmurHash3.h -> build\lib.win32-cpython-311\murmurhash\include\murmurhash
      running build_ext
      building 'murmurhash.mrmr' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]
  
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for murmurhash
    Failed to build cymem murmurhash
    ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (cymem, murmurhash)
    [end of output]
  
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: subprocess-exited-with-error
  
  pip subprocess to install build dependencies did not run successfully.
  exit code: 1
  
  See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

pip subprocess to install build dependencies did not run successfully.
exit code: 1

See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by zyzy5** [comment]
calibre, version 7.15.0 (win32, embedded-python: True)
Joint MOBI: 请使用 KindleUnpack 的 "-s " 选项来拆分这本书。

Starting job: 正在为《白痴：全2册（陀思妥耶夫斯基蜚声世界文坛的优秀作品之一；经典“网格本”重新上市） (外国文学名著丛书)》生成生词提示和X-Ray 
Job: "正在为《白痴：全2册（陀思妥耶夫斯基蜚声世界文坛的优秀作品之一；经典“网格本”重新上市） (外国文学名著丛书)》生成生词提示和X-Ray" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 85, in start_work
  File "calibre_plugins.worddumb.parse_job", line 129, in do_job
  File "calibre_plugins.worddumb.metadata", line 202, in get_asin_etc
  File "calibre_plugins.worddumb.metadata", line 222, in extract_mobi
Exception: JointMOBI
 
Called with args: (ParseJobData(book_id=108, book_path='C:\\Users\\h5482\\Calibre1997\\Tuo Si Tuo Ye Fu Si Ji\\Bai Chi _Quan 2Ce (Tuo Si Tuo Ye Fu (108)\\Bai Chi _Quan 2Ce (Tuo Si Tuo Y - Tuo Si Tuo Ye Fu Si Ji_en.mobi', mi=<calibre.ebooks.metadata.book.base.Metadata object at 0x000002045F4A0250>, book_fmt='MOBI', book_lang='zh', useragent='', plugin_path='', spacy_model='', create_ww=True, create_x=True, asin='BBTKKL356W', acr='Bai_Chi__Quan_2Ce_(Tuo_Si_Tuo_Y', revision='173e3c14', kfx_json=None, mobi_html=b'', mobi_codec='utf-8'),) {'notifications': <queue.Queue object at 0x0000020464EB9390>, 'abort': <threading.Event at 0x20464eb9710: unset>, 'log': <calibre.utils.logging.GUILog object at 0x0000020464EB99D0>}

**Reply by xxyzz** [reply]
You need to split this book as the error message says.

**Comment by zyzy5** [comment]
python 3.11.8win32
pip 24.2

**Comment by xxyzz** [comment]
You need to install 64bit Python.

**Reply by zyzy5** [reply]
thanks, it works
